### ISSUE TITLE
Conditional rendering in view!

### ISSUE THREAD

**Issue by m0rtis0** [Relm4/Relm4#164]
Is it currently possible to conditionally render different Widgets in the view! macro?
According to https://relm4.org/book/widget_macro_reference.html#functions one can use a special syntax to specify a function that returns the widget. Is it possible to use a closure here?
And additionally: How do I write conditional logic without losing the macro abilities? (Can I "pass" the macro to the function that conditionally renders my widget?)

**Comment by AaronErhardt** [comment]
So far, you can't use closures, but if you have a good idea for a syntax you can add it here: https://github.com/Relm4/Relm4/issues/162

Regarding conditional code, you can use `?` for optional assignment of properties that are stores as `Option<T>`. Yet, you can't generate widgets conditionally yet. Again, if you have a good idea for a syntax, let us know :)

**Comment by m0rtis0** [comment]
Hey there, thanks for you response! Will do that.

**Comment by euclio** [comment]
I do conditional rendering by setting the visibility on the different widgets depending on what's in the model: https://github.com/euclio/fm/blob/9f1842c9e25a29022784595144c0311ef5a6f8f5/src/file_preview.rs#L272-L294

**Comment by AaronErhardt** [comment]
Yes, that'd be one solution. But since we need to generate a hidden container anyway, I thought using a Stack might be a better option and would also allow using animations on page switches. I hope I can soon add this to #165
